### PR TITLE
Update dependencies and add tests for empty file field

### PR DESCRIPTION
**Pull request by uncle-lv**
<del>**Pin flask-marshmallow==1.2.1** for two important bug fixes below:</del>

- [Fix File field when it receives an empty value](https://github.com/marshmallow-code/flask-marshmallow/pull/301)
- [Fix validate.FileSize to handle SpooledTemporaryFile](https://github.com/marshmallow-code/flask-marshmallow/pull/300)

<del>**Pin flask==2.2.0 and werkzeug==2.2.0** because [flask-marshmallow updated min version of flask](https://github.com/marshmallow-code/flask-marshmallow/pull/289).</del>

<del>**Pin Flask-HTTPAuth==4.4.0** because [`safe_str_cmp` is deprecated and has been removed in `Werkzeug 2.1`](https://github.com/miguelgrinberg/Flask-HTTPAuth/pull/126).</del>

**Update flake8 hook to 7.0.0** to be [compatible with `python3.12`](https://github.com/PyCQA/flake8/issues/1701).

Add a test for empty file field.

### CODE REVIEW

File: requirements/min-versions.in [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # minimum install requirements
```
Comment (greyli) [inline requirements/min-versions.in:1]: You don't need to update this file. This file kept the minimal version supported by APIFlask and will only be used in testing.

**Review by greyli — CHANGES_REQUESTED** [review]

File: requirements/min-versions.in (hunk 1 above)
Comment (uncle-lv) [reply requirements/min-versions.in:1]: If I don't update this file, the new test will fail.
```python
        rv = client.post(
            '/',
            data={
                'image': '',
            },
            content_type='multipart/form-data'
        )
>       assert rv.status_code == 200
E       assert 422 == 200
E         +422
E         -200

tests/test_fields.py:54: AssertionError
```

Comment (greyli) [reply requirements/min-versions.in:1]: You can skip this test based on the version of flask-marshmallow. Example: https://github.com/apiflask/apiflask/blob/4992f8efe0c22025f67eadd366891f25e3a152e6/examples/test_examples.py#L118

Comment (uncle-lv) [reply requirements/min-versions.in:1]: `flask_marshmallow` has no attribute `__version__`. I'll make a pr for it.
```python
AttributeError: module 'flask_marshmallow' has no attribute '__version__'
```

Comment (greyli) [reply requirements/min-versions.in:1]: Use `importlib.metadata`:

```
>>> from importlib.metadata import version
>>> version('flask')
'2.3.3'
>>> version('flask-marshmallow')
'0.15.0'
```

Comment (greyli) [reply requirements/min-versions.in:1]: BTW, could you help update the reference of `flask_sqlalchemy.__version__` and change to use `importlib.metadata.version`?

[That variable is deprecated](https://github.com/pallets-eco/flask-sqlalchemy/blob/main/src/flask_sqlalchemy/__init__.py#L12-L26).

**Comment by uncle-lv** [comment]
> Why did you revert the changes to the requirement files? It's fine to update them (except the `min-versions.*`).

Because in the end no dependencies are updated.

**Review by greyli — APPROVED** [review]
